hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackHelper.java

```diff
@@ -116,8 +118,10 @@ List<Pair<String, HoodieRollbackStat>> maybeDeleteAndCollectStats(HoodieEngineCo
                                                                     HoodieInstant instantToRollback,
                                                                     List<SerializableHoodieRollbackRequest> rollbackRequests,
                                                                     boolean doDelete, int numPartitions) {
+    List<SerializableHoodieRollbackRequest> groupedRollbackRequests =
+        groupSerializableRollbackRequestsBasedOnFileGroup(rollbackRequests);
     final TaskContextSupplier taskContextSupplier = context.getTaskContextSupplier();
-    return context.flatMap(rollbackRequests, (SerializableFunction<SerializableHoodieRollbackRequest, Stream<Pair<String, HoodieRollbackStat>>>) rollbackRequest -> {
+    return context.flatMap(groupedRollbackRequests, (SerializableFunction<SerializableHoodieRollbackRequest, Stream<Pair<String, HoodieRollbackStat>>>) rollbackRequest -> {
```

Comment:
can we try to limit the scope of the changes only to tbl version 6. 
in tbl version 8, we don't need this since log files are going to be deleted during rollback execution.

Reply:
Fixed.